genesyscloud/authorization_product/data_source_genesyscloud_authorization_product.go

```diff
@@ -0,0 +1,65 @@
+package authorization_product
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	gcloud "terraform-provider-genesyscloud/genesyscloud"
+)
+
+func DataSourceAuthorizationProduct() *schema.Resource {
+	return &schema.Resource{
+		Description: `Data source for Genesys Cloud Authorisation Products.`,
+
+		ReadContext: gcloud.ReadWithPooledClient(dataSourceAuthorizationProductRead),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: `Authorization Product name.`,
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+		},
+	}
+}
```

Comment:
Could this be moved to a `_schema.go` file like with other packages? Just to keep with the format